src/index.ml

```diff
@@ -296,6 +298,83 @@ struct
           Int64.of_float rounded
       end)
 
+  let try_load_log t path =
+    Log.debug (fun l ->
+        l "[%s] checking on-disk %s file" (Filename.basename t.root)
+          (Filename.basename path));
+    if Sys.file_exists path then (
+      let io =
+        IO.v ~fresh:false ~readonly:true ~generation:0L ~fan_size:0L path
+      in
+      let mem = Tbl.create 0 in
+      iter_io (fun e -> Tbl.replace mem e.key e.value) io;
```

Comment:
Why do you need `replace` here?  Would `add` also work? If yes we should be using `add` as it is much more efficient.